client/my-sites/site-settings/form-general.jsx

```diff
@@ -127,35 +120,68 @@ const FormGeneral = React.createClass( {
 		return this.recordEventOnce.bind( this, key, eventAction );
 	},
 
+	setDirtyField( key ) {
+		const newState = {};
+		const dirtyFields = this.state.dirtyFields || [];
+		if ( dirtyFields.indexOf( key ) === -1 ) {
+			newState.dirtyFields = [ ...dirtyFields, key ];
+		}
+		this.setState( newState );
+	},
+
+	handleChange( event ) {
+		const currentTargetName = event.currentTarget.name,
+			currentTargetValue = event.currentTarget.value;
+
+		this.setDirtyField( currentTargetName );
+		this.setState( { [ currentTargetName ]: currentTargetValue } );
+	},
+
+	handleToggle( name ) {
```

Comment:
@ryelle I was thinking the same thing! We should definitely do that, but let's do it in another PR after we have both these merged.

@omarjackman I'm fine both ways, but just so you know I've reused the same code that was implemented by @ryelle in #10167. Once we move these into `form-base`, we can consider renaming them.